scanpy/preprocessing/_highly_variable_genes.py

```diff
@@ -298,7 +489,12 @@ def highly_variable_genes(
     max_mean: Optional[float] = 3,
     span: Optional[float] = 0.3,
     n_bins: int = 20,
-    flavor: Literal['seurat', 'cell_ranger', 'seurat_v3'] = 'seurat',
+    theta: float = 100,
```

Comment:
I think all the new args introduced by pearson residuals should be `Optional` ?

Reply:
I'm new to the typing system and this was something I was not sure about - when I read up on it [here](https://docs.python.org/3/library/typing.html), I found this:

>`Optional[X]` is equivalent to `Union[X, None]`.
>Note that this is not the same concept as an optional argument, which is one that has a default. An optional argument >with a default does not require the `Optional` qualifier on its type annotation just because it is optional. For example:
```
def foo(arg: int = 0) -> None:
    ...
```
>On the other hand, if an explicit value of `None` is allowed, the use of `Optional` is appropriate, whether the argument is optional or not. For example:
```
def foo(arg: Optional[int] = None) -> None:
    ...
```

I thought the newly introduced arguments here are more of the first case, because setting them explicitly to `None` is not allowed in certain situations (i.e. will cause an error when `flavor='pearson_residuals'`), so I decided not to use `Optional` here (but I see now the same logic would apply to `min_disp` etc and span`)

What do you think?

Reply:
> but I see now the same logic would apply to min_disp etc and span

exactly yes, I think `Optional` is more appropriate if the argument is supposed to work only on a subset of function behavior. I'm also not super strong on this so some people might have different opinion